work16/src/main.js

```diff
@@ -0,0 +1,251 @@
+
+  span.classList = "newIcon";
+  span.textContent = "NEW"
+
+  return span;
+}
+
+const createCommentIcon = (commentArray) => {
+  const div = document.createElement("div");
+  const spanLayers = document.createElement("span");
+  const i = document.createElement("i");
+  const spanCounter = document.createElement("span");
+
+  div.classList = "commentIcon";
+  spanLayers.classList = "fa-layers fa-fw";
+  i.classList = "fa fa-light fa-comment faa-vertical animated";
+  spanCounter.classList = "fa-layers-counter";
+
+  spanCounter.textContent = `${commentArray.length}`;
+
+  spanLayers.appendChild(i);
+  spanLayers.appendChild(spanCounter)
+  div.appendChild(spanLayers);
+
+  return div;
+}
+
+const createSection = (tabTopics, index) => {
+  const section = document.createElement("section");
+  const tabTopicElem = document.getElementById(tabTopics[index].id);
+  const tabIdName = tabTopicElem.id;
+
+  section.id = `${"tabpanelTopics" + (index + 1)} `;
+  section.setAttribute("aria-labelledby", tabIdName);
+  section.setAttribute("roll", "tabpanel")
+
+  return section;
+}
+
+const createTopicImg = (imgPath) => {
+  const img = document.createElement("img");
+  const div = document.createElement("div");
+
+  div.classList = "tabTopicImg";
+  img.src = imgPath;
+  img.width = "350";
+  img.height = "250";
+
+  div.appendChild(img);
+
+  return div;
+}
+
+const hideLoadingImg = () => {
+  document.getElementById("js_loading_wrap").remove();
+}
+
+const isLatestArticles = (newsArticleData) => {
+  const PERIODOFLATESTARTICLES = 3;
+  const newsArticleDate = newsArticleData.date;
+
+  const nowDate = new Date();
+  const postDate = new Date(newsArticleDate);
+
+  const periodOfDays = differenceInCalendarDays(nowDate, postDate);
+  const result = periodOfDays <= PERIODOFLATESTARTICLES;
+
+  return result;
+}
+
+const removeTabPanel = () => {
+  document.getElementById("js_tabpanelTopics_wrap").textContent = "";
+}
+
+const clickedTabEvent = (newsDataArray) => {
+  newsDataArray.forEach((newsCategoryObj, newsCategoryIndex) => {
+    const tabTopics = document.querySelectorAll(".tabTopics");
+    tabTopics[newsCategoryIndex].addEventListener("click", (event) => {
+
+      removeTabPanel();
+
+      const sectionElem = document.querySelector("section");
+      const tabpanelTopicsWrap = document.querySelector(".tabpanelTopics_wrap");
+      const selectedTab = document.querySelector('[aria-selected="true"]');
+
+      const clickedArticleContent = newsDataArray[newsCategoryIndex].contents;
+      const clickedTopicImg = newsCategoryObj.img
+
+      sectionElem.id = `tabpanelTopics${newsCategoryIndex + 1}`;
+      sectionElem.setAttribute("aria-labelledby", `js_tabTopics${newsCategoryIndex + 1}`)
+      selectedTab.ariaSelected = false;
+      event.currentTarget.ariaSelected = true;
+
+      tabpanelTopicsWrap.prepend(createArticle(clickedArticleContent));
+      tabpanelTopicsWrap.appendChild(createTopicImg(clickedTopicImg))
+    });
+  });
```

Comment:
IMO

```suggestion
const clickedTabEvent = (newsDataArray) => {
  const tabTopics = [...document.querySelectorAll(".tabTopics")];
  tabTopics.forEach((tab) => {
    tab.addEventListener("click", (event) => {
      removeTabPanel();
      const selectedTab = document.querySelector('[aria-selected="true"]');
      selectedTab.ariaSelected = false;
      event.currentTarget.ariaSelected = true;

      const index = tabTopics.indexOf(event.currentTarget);
      const tabpanelTopicsWrap = document.querySelector(".tabpanelTopics_wrap");
      tabpanelTopicsWrap.prepend(createArticle(newsDataArray[index].contents));
      tabpanelTopicsWrap.appendChild(createTopicImg(newsDataArray[index].img));

      const sectionElem = document.querySelector("section");
      sectionElem.setAttribute("aria-labelledby", `js_tabTopics${index + 1}`);
    })
  })
};
```

Reply:
The way to write the [...] is called spread syntax...
You are using spread syntax to convert it to an array.
I am embarrassed to say that I learned that querySelectorAll returns a NodeList.
Also, I don't have to write in a newsDataArray foreach statement, so the code is easier to read.

I had put variables in the function arguments, but it's easier to read without them... I learned a lot.
I had written const all together, but changing the order of execution made it easier to understand.

Thank you very much.

commit: [9a0fff9](https://github.com/yasu-pro/js-handson/pull/18/commits/9a0fff93a90da6b4f7386b943965a9687fd12114)